tests/listobject_test.py

```diff
@@ -1,32 +1,205 @@
-                                            aws_access_key_id="no",
-                                            aws_secret_access_key="heck")
-        set_access("", 'own',recursive=True)
-        mkdir("", access_level="own")
+                                            endpoint_url=self.s3_bridge_url,  # normal networking stuff :p
+                                            aws_access_key_id=self.key,
+                                            aws_secret_access_key=self.secret_key)
 
     def tearDown(self):
-        remove_file(self.client, "", recursive=True)
+        assert_command('irm -rf %s/f1 %s/f2 %s/dir1 %s/dir2' % (self.bucket_irods_path,
+            self.bucket_irods_path, self.bucket_irods_path, self.bucket_irods_path))
+        assert_command('rm f1')
+        assert_command('rm f2')
+
+    # ======== Helper Functions =========
+
+    # used to assert keys are in the contents list returned by botocore
+    # possibly checking the size and LastModified time.
+    def assert_key_in_contents_list(self, list_objects_result, key, size=None, lastmodified=None):
+        contents_list = list_objects_result['Contents']
+        matching_key = None
+        matching_size = None
+        matching_lastmodified = None
+        for entry in contents_list:
+            if entry['Key'] == key:
+                matching_key = entry['Key']
+                matching_size = entry['Size']
+                matching_lastmodified = entry['LastModified']
+    
+        self.assertIsNotNone(matching_key, 'Key not found [%s]' % key)
+        self.assertIsNotNone(matching_size, 'Size not found for key [%s]' % key)
+        self.assertIsNotNone(matching_lastmodified, 'LastModified is not found for key %s' % key)
+        if size != None:
+            self.assertEqual(matching_size, size, 'Size does not match for key %s' % key)
+        if lastmodified != None:
+            # Only checking year/month/day.  This could fail if the time between
+            # writing the file to iRODS and  getting the current datetime object
+            # rolled over to a new day.
+            self.assertEqual(matching_lastmodified.year, lastmodified.year, 'Year does not match for key %s' % key) 
+            self.assertEqual(matching_lastmodified.month, lastmodified.month, 'Month does not match for key %s' % key)
+            self.assertEqual(matching_lastmodified.day,  lastmodified.day, 'Day does not match for key %s' % key)
+    
+    # used to assert keys are in the CommonPrefixes list returned by botocore
+    def assert_prefix_in_common_prefixes_list(self, list_objects_result, prefix):
+        common_prefixes_list = list_objects_result['CommonPrefixes']
+        matching_key = None
+        for entry in common_prefixes_list:
+            if entry['Prefix'] == prefix:
+                matching_key = entry['Prefix']
+        self.assertIsNotNone(matching_key, 'Prefix is not found for prefix=%s' % prefix)
```

Comment:
You could make this part of the if-stmt's body and then break.

Consider rewording the message. For example:
```
Prefix not found [prefix=%s]
```
or
```
Prefix [%s] not found
```

If the assertion fails, the default message is probably okay too. Up to you.